### DISCUSSION TITLE
macOS directory permission issues after using Next installation

### DISCUSSION THREAD

**Discussion by Patrikur** [vercel/next.js#68153] [Help]
### Summary

Hi people,

Since I started using a Mac I always get permission errors for my directory after installing a new Next project and I want to save changes to any file. Of course, I run the installation with `sudo`, but even then I get permission issues.
I found the classic fix by changing local permission with `sudo chown -Rv username path`, but is there any setting I am overseeing or forgot that I do not have to do this and have permissions everywhere automatically?

Thanks in advance!

**Comment by Juneezee** [comment]
You don't need to run with `sudo`. If you use `sudo`, all folders and files created will be owned by `root`. This is why you get those permission issues.

**Reply by Patrikur** [reply]
When I do not use `sudo` I am getting this error:
`Aborting installation.

Unexpected error. Please report it as a bug:
 Error: EACCES: permission denied, open '/Users/x/Library/Preferences/create-next-app-nodejs/config.json.tmp-1903785941703c75'
    at openSync (node:fs:603:3)
    at attempt (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:8:43165)
    at Object.writeFileSync (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:8:38664)
    at Conf._write (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:8:74912)
    at set store [as store] (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:8:73358)
    at Conf.set (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:8:71962)
    at run (/Users/x/.npm/_npx/cc2145a2fe1558fa/node_modules/create-next-app/dist/index.js:74:51291)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -13,
  syscall: 'open',
  code: 'EACCES',
  path: '/Users/x/Library/Preferences/create-next-app-nodejs/config.json.tmp-1903785941703c75'
}
`

**Reply by Juneezee** [reply]
My guess is that your `node`, `npm` are not installed correctly.

Did you install through a Node version manager such as `nvm`? See https://docs.npmjs.com/downloading-and-installing-node-js-and-npm for more info

**Reply by Patrikur** [reply]
Yes, also double checked and updated to be sure, still getting the same error.

**Reply by Juneezee** [reply]
Could you try the steps here and see if it works? https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally#manually-change-npms-default-directory

**Comment by icyJoseph** [comment]
Hi,

This looks like `create-next-app` is having issues saving your prompt preferences. 

I wonder if your terminal has permission to the Preferences folder? Which terminal emulator do you use?

**Reply by Patrikur** [reply]
The terminal emulator is `xterm-256color`.
In the preferences folder I can find `create-next-app-nodejs`, or do I need to dig deeper?

**Reply by icyJoseph** [reply]
Well, I'd guess that folder is empty somehow? Because `Error: EACCES: permission denied, open '/Users/x/Library/Preferences/create-next-app-nodejs/config.json.tmp-1903785941703c75'` is claiming that it couldn't access such a file

At some level, a process is missing permissions to access these files, 🤔 

- https://apple.stackexchange.com/a/374142
